### PR TITLE
add a new parameter allow_roleassignment_on_rg to allow/disallow role assignment on resource group level

### PR DESCRIPTION
**Pull request by wenjie1070116**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py
Comment (debuggerXi) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py:1]: miss allowOnRP

**Review by debuggerXi — APPROVED** [review]
all good

**Review by matferrari-msft — APPROVED** [review]

**Review by debuggerXi — CHANGES_REQUESTED** [review]
allow_roleassignment_on_rg missing in many places

**Review by luigiw — APPROVED** [review]

**Review by MilesHolland — COMMENTED** [review]
Would it be possible to add or amend an existing workspace e2e test to check that this new parameter is handled properly in live scenarios? The test_workspace_create_and_delete test would be a good place to try assigning this new value.

**Pull request merged**